### ISSUE TITLE
I can not compile RxSwift on 3.0.0.alpha.1

### ISSUE THREAD

**Issue by shiratsu** [ReactiveX/RxSwift#881]
Hi,there.

I tried to use RxSwift on swift3.
And I use carthage.
When i got library by carthage,I got the following error.
```
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Schedulers/SchedulerServices+Emulation.swift:32:19: error: assigning non-escaping parameter 'action' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/AnyObserver.swift:35:25: error: assigning non-escaping parameter 'eventHandler' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/WithLatestFrom.swift:114:27: error: assigning non-escaping parameter 'resultSelector' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/Zip+arity.swift:95:27: error: assigning non-escaping parameter 'resultSelector' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/Zip+arity.swift:196:27: error: assigning non-escaping parameter 'resultSelector' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/Zip+arity.swift:305:27: error: assigning non-escaping parameter 'resultSelector' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/Zip+arity.swift:422:27: error: assigning non-escaping parameter 'resultSelector' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/Zip+arity.swift:547:27: error: assigning non-escaping parameter 'resultSelector' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/Zip+arity.swift:680:27: error: assigning non-escaping parameter 'resultSelector' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/Zip+arity.swift:821:27: error: assigning non-escaping parameter 'resultSelector' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/Deferred.swift:53:30: error: assigning non-escaping parameter 'observableFactory' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/Zip+CollectionType.swift:128:31: error: assigning non-escaping parameter 'resultSelector' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/Catch.swift:86:20: error: assigning non-escaping parameter 'handler' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/CombineLatest.swift:114:27: error: assigning non-escaping parameter 'setLatestValue' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/SkipWhile.swift:93:22: error: assigning non-escaping parameter 'predicate' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/SkipWhile.swift:100:31: error: assigning non-escaping parameter 'predicate' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/DistinctUntilChanged.swift:61:21: error: assigning non-escaping parameter 'selector' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/DistinctUntilChanged.swift:62:21: error: assigning non-escaping parameter 'comparer' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Schedulers/VirtualTimeScheduler.swift:273:23: error: assigning non-escaping parameter 'action' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/Multicast.swift:62:28: error: assigning non-escaping parameter 'subjectSelector' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/Multicast.swift:63:21: error: assigning non-escaping parameter 'selector' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/CombineLatest+CollectionType.swift:116:27: error: assigning non-escaping parameter 'resultSelector' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observers/AnonymousObserver.swift:22:25: error: assigning non-escaping parameter 'eventHandler' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/Merge.swift:182:21: error: assigning non-escaping parameter 'selector' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/Merge.swift:198:21: error: assigning non-escaping parameter 'selector' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/Merge.swift:219:21: error: assigning non-escaping parameter 'selector' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/Merge.swift:362:21: error: assigning non-escaping parameter 'selector' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/Merge.swift:381:21: error: assigning non-escaping parameter 'selector' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/Merge.swift:401:21: error: assigning non-escaping parameter 'selector' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/CombineLatest+arity.swift:83:27: error: assigning non-escaping parameter 'resultSelector' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/CombineLatest+arity.swift:171:27: error: assigning non-escaping parameter 'resultSelector' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/CombineLatest+arity.swift:266:27: error: assigning non-escaping parameter 'resultSelector' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/CombineLatest+arity.swift:368:27: error: assigning non-escaping parameter 'resultSelector' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/CombineLatest+arity.swift:477:27: error: assigning non-escaping parameter 'resultSelector' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/CombineLatest+arity.swift:593:27: error: assigning non-escaping parameter 'resultSelector' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/CombineLatest+arity.swift:716:27: error: assigning non-escaping parameter 'resultSelector' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Schedulers/RecursiveScheduler.swift:26:19: error: assigning non-escaping parameter 'action' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Schedulers/RecursiveScheduler.swift:138:19: error: assigning non-escaping parameter 'action' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/Scan.swift:56:24: error: assigning non-escaping parameter 'accumulator' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/AnonymousObservable.swift:48:29: error: assigning non-escaping parameter 'subscribeHandler' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Disposables/AnonymousDisposable.swift:36:26: error: assigning non-escaping parameter 'disposeAction' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Disposables/AnonymousDisposable.swift:42:26: error: assigning non-escaping parameter 'disposeAction' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/Switch.swift:152:21: error: assigning non-escaping parameter 'selector' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/Switch.swift:185:21: error: assigning non-escaping parameter 'selector' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/Map.swift:20:21: error: assigning non-escaping parameter 'selector' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/Map.swift:57:21: error: assigning non-escaping parameter 'selector' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/Map.swift:91:21: error: assigning non-escaping parameter 'selector' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/Map.swift:114:21: error: assigning non-escaping parameter 'selector' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/Using.swift:69:28: error: assigning non-escaping parameter 'resourceFactory' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/Using.swift:70:30: error: assigning non-escaping parameter 'observableFactory' to an @escaping closure
/Users/xxxxxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/TakeWhile.swift:111:22: error: assigning non-escaping parameter 'predicate' to an @escaping closure
/Users/xxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/TakeWhile.swift:118:31: error: assigning non-escaping parameter 'predicate' to an @escaping closure
/Users/xxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/Amb.swift:33:17: error: assigning non-escaping parameter 'sink' to an @escaping closure
/Users/xxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/Do.swift:47:25: error: assigning non-escaping parameter 'eventHandler' to an @escaping closure
/Users/xxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/Reduce.swift:65:24: error: assigning non-escaping parameter 'accumulator' to an @escaping closure
/Users/xxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/Reduce.swift:66:22: error: assigning non-escaping parameter 'mapResult' to an @escaping closure
/Users/xxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/Filter.swift:18:22: error: assigning non-escaping parameter 'predicate' to an @escaping closure
/Users/xxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/Filter.swift:50:22: error: assigning non-escaping parameter 'predicate' to an @escaping closure
/Users/xxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Schedulers/Internal/ScheduledItem.swift:26:19: error: assigning non-escaping parameter 'action' to an @escaping closure
/Users/xxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/RxSwift/Observables/Implementations/Zip.swift:126:25: error: assigning non-escaping parameter 'setNextValue' to an @escaping closure
A shell task (/usr/bin/xcrun xcodebuild -workspace /Users/xxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/Rx.xcworkspace -scheme RxBlocking-iOS -configuration Release -sdk iphoneos ONLY_ACTIVE_ARCH=NO BITCODE_GENERATION_MODE=bitcode CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES clean build) failed with exit code 65:
2016-09-14 14:56:22.739 xcodebuild[12815:667003] [MT] DVTPlugInManager: Required plug-in compatibility UUID 8A66E736-A720-4B3C-92F1-33D9962C69DF for KSImageNamed.ideplugin (com.ksuther.KSImageNamed) not present
--- xcodebuild: WARNING: Unable to open project file '/Users/xxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/Rx.playground' in workspace '/Users/xxxxxx/swift3/xxxxxxxxx/Carthage/Checkouts/RxSwift/Rx.xcworkspace'.
** BUILD FAILED **
```

My Cartfile is the following code.
```
github "ReactiveX/RxSwift" "3.0.0.alpha.1"
```
And the command of carthage is here.
```
carthage update --platform ios --no-use-binaries
```

Is this error the same with https://github.com/ReactiveX/RxSwift/issues/880?

By the way,
At first I set to Cartfile the following code.
```
github "ReactiveX/RxSwift" "3.0.0-beta.1"
```
But i got the following error.
```
Failed to check out repository into /Users/xxxxxx/Library/Caches/org.carthage.CarthageKit/dependencies/RxSwift: No object named "3.0.0-beta.1" exists
```

**Comment by tcurdt** [comment]
For now I would use `github "ReactiveX/RxSwift" "develop"` and let's hope there is a new release soon.

**Comment by shiratsu** [comment]
@tcurdt Oh,Thank you.I will also try to use develop branch.

**Comment by kzaher** [comment]
Hi @shiratsu ,

we've just published 3.0.0-beta.1 version that compiles with Xcode Version 8.0 (8A218a).
